src/controllers/user-login-controller.js

```diff
@@ -0,0 +1,60 @@
+import passport from 'passport';
+import { generateToken } from '../utils/generateToken';
+import { User } from '../database/models';
+import { BcryptUtil } from '../utils/bcrypt';
+
+const loginUser = async (req, res, next) => {
+  passport.authenticate('local', async (err, user) => {
+    if (err) {
+      return next(err);
+    }
+
+    if (!user) {
+      return res.status(401).json({ message: 'Invalid email or password' });
+    }
+
+    try {
+      const foundUser = await User.findOne({ where: { email: user.email } });
+      if (!foundUser) {
+        return res.status(404).json({ message: 'User not found' });
+      }
+
+      const passwordMatches = await BcryptUtil.compare(req.body.password, user.password);
+
+      if (!passwordMatches) {
+        return res.status(401).json({ message: 'Invalid email or password' });
+      }
+      const UserToken = {
+        id: user.id,
+        firstname: user.firstname,
+        lastname: user.lastname,
+        email: user.email,
+        role: user.role,
+        isActive: user.isActive,
+        mustUpdatePassword: user.mustUpdatePassword,
+        lastTimePasswordUpdated: user.lastTimePasswordUpdated
+      };
+      const token = generateToken(UserToken);
+
+      return res.status(200).json({
+        message: 'Successful login',
+        user: {
+          id: foundUser.id,
+          firstname: foundUser.firstname,
+          lastname: foundUser.lastname,
+          email: foundUser.email,
+          role: foundUser.role,
+        },
+        token: token,
+      });
+    } catch (error) {
+      return next(error);
+    }
+  })(req, res, next);
+};
+
+const protectedRouteHandler = async (req, res, user) => {
+  res.json({ message: `welcome, ${user.firstname} ` });
```

Comment:
what will happen if an invalid user passed in this func?
it's better to catch the errors that might happen.

`const protectedRouteHandler = async (req, res, user) => {
  try {
    res.json({ message: `welcome, ${user.firstname} ` });
  } catch (error) {
    return res.status(500).json({ message: 'Internal server error' });
  }
};`